app/models/garden.rb

```diff
@@ -1,5 +1,7 @@
 class Garden < ApplicationRecord
 	# validations
+  validates_presence_of :name, :latitude, :longitude
+  validates_numericality_of :latitude, :longitude, :max_moisture, :min_moisture
 
```

Comment:
@Mycobee Thoughts on adding conditional validation that min_moisture has to be present if auto_water is set to true?

Reply:
Done in the new changes 👍